### DISCUSSION TITLE
drag events for IText

### DISCUSSION THREAD

**Discussion by ShaMan123** [fabricjs/fabric.js#7800] [Ideas]
@asturur @melchiar 
I'm in need for drag event support for text so it behaves more like a textarea.
Specifically dragging selected text.
What are your thoughts? Should it be supported by fabric?

**Comment by melchiar** [comment]
The ability to select and drag/drop text within an itext/textbox object would probably be unobtrusive enough of an addition to justify official support. 

If you wanted to fully match the behavior of textareas though you'd need to also support dragging selected text between different objects. I don't think that would be doable without some pretty major rewrites to Fabric's object interaction logic since you'd essentially need to allow more than one object to be in editing mode at a time.

![text-area-dragging](https://user-images.githubusercontent.com/9057412/157996261-61e07ed3-9c3f-4e03-a095-ba961a5f20df.gif)

**Comment by ShaMan123** [comment]
#7802
